providers/kafka_provider.ts

```diff
@@ -1,64 +1,29 @@
 import { ApplicationService, KafkaConfig } from '@adonisjs/core/types'
 
 import { Kafka } from '../src/index.ts'
-import { Producer } from '../src/producer.ts'
-
-import { HttpContext } from '@adonisjs/core/http'
-
-declare module '@adonisjs/core/http' {
-  export interface HttpContext {
-    kafka: {
-      producers: {
-        [key: string]: Producer
-      }
-    }
-  }
-}
 
 export default class KafkaProvider {
   private app: ApplicationService
-  private config: KafkaConfig
 
   constructor(app: ApplicationService) {
     this.app = app
-    this.config = this.app.config.get<KafkaConfig>('kafka')
   }
 
   register() {
     this.app.container.singleton('kafka', async () => {
       const logger = await this.app.container.make('logger')
-      return new Kafka(logger, this.config)
+      const config = this.app.config.get<KafkaConfig>('kafka')
+      return new Kafka(logger, config)
```

Comment:
This is how other modules I looked at did config, rather than a `this.config` which we only needed for the enabled stuff.